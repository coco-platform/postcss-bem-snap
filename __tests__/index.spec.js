/**
 * @description - UT BEM plugin
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// package
const fs = require('fs');
const path = require('path');
const postcss = require('postcss');
// internal
const bem = require('../');
// scope
const snapshot = path.resolve(__dirname, '__fixture__', 'bem.pcss');
const content = fs.readFileSync(snapshot, { encoding: 'utf8' });

describe('postcss-bem-snap', () => {
  it('should complete standard workflow', () => {
    const plugins = [bem()];

    return postcss(plugins)
      .process(content, { from: '' })
      .then((result) => {
        expect(result.css).toMatchSnapshot();
        expect(result.warnings().length).toBe(0);
      });
  });
});
