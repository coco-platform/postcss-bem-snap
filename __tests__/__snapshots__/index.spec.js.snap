// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss-bem-snap should complete standard workflow 1`] = `
".vip {
    text-align: center;
    background-color: aquamarine
}
.vip--platinum {
    background-color: plum
}
.vip--gold {
    background-color: gold
}
.vip__header {
    color: mediumpurple
}
.vip__header--lg {
    font-size: 36px
}
.vip__header--md {
    font-size: 24px
}
.vip__header--sm {
    font-size: 16px
}
"
`;
